helm/minio/templates/_helper_create_user.txt

```diff
@@ -97,7 +97,7 @@ echo {{ tpl .accessKey $global }} > $MINIO_ACCESSKEY_SECRETKEY_TMP
 {{- if .existingSecret }}
 cat /config/secrets/{{ tpl .existingSecret $global }}/{{ tpl .existingSecretKey $global }} >> $MINIO_ACCESSKEY_SECRETKEY_TMP
 # Add a new line if it doesn't exist
-sed -i '$a\' $MINIO_ACCESSKEY_SECRETKEY_TMP
+echo >> $MINIO_ACCESSKEY_SECRETKEY_TMP
```

Comment:
Wait how does it fix usage of sed?

Reply:
Here, only the method of command replacement is used to make it functional.
I apologize for the misunderstanding.Due to the fact that the quay.io/minio/mc:RELEASE.2023-09-29T16-41-22Z image does not support the sed command, can also use the following two methods.
1、Roll back the base image to use ubi-minimal again.
2、Install sed in the mc container image, based on ubi-micro.

Reply:
The purpose of the sed command is to insert a blank line at the end of the target file. If the last line is absent in the target file, the command will create a new line at the file's conclusion.

Alternatively, you can achieve the same result using the echo command along with the >> operator. This approach effectively adds a blank line to the end of the file.

Reply:
and why do we need that blankline?

Reply:
I used echo just to address the issue of the sed command not being available in the image, causing it to be unusable.
![image](https://github.com/minio/minio/assets/39941928/89e4c67f-129a-46eb-930e-df6e691e6a42)

For the above repair method, the echo command is indeed not necessary.

Reply:
I used echo just to address the issue of the sed command not being available in the image, causing it to be unusable.
![1700131361038](https://github.com/minio/minio/assets/112474703/4353a495-8ec9-4b33-a163-9d9398bea08c)
For the above repair method, the echo command is indeed not necessary.

Reply:
@harshavardhana 
for this 
```shell
cat /config/secrets/{{ tpl .existingSecret $global }}/{{ tpl .existingSecretKey $global }} >> $MINIO_ACCESSKEY_SECRETKEY_TMP
# Add a new line if it doesn't exist
sed -i '$a\' $MINIO_ACCESSKEY_SECRETKEY_TMP
```
before the sed action.
cat will add the environment like
```env
export A=B
```
if we not do this, we will get 
```env
export A=BOTHERS
```

Reply:
@jiuker as per my testing `sed -i '$a\'` doesn't have a real effect.

Reply:
> as per my testing `sed -i '$a\'` doesn't have a real effect.

It depends on whether your secret configuration comes with a built-in carriage return @vadmeste